modules/createHashHistory.js

```diff
@@ -33,7 +37,7 @@ var DefaultQueryKey = '_k';
 function createHashHistory(options={}) {
   var { queryKey } = options;
 
-  if (queryKey && typeof queryKey !== 'string')
+  if (typeof queryKey !== 'string')
```

Comment:
Ah, I see why you did this now. My original intent was to let people use hash history without a `queryKey`. They wouldn't be able to use state, but they would have somewhat cleaner URLs too. So it's a tradeoff.